app/frontend/src/features/messages/groupchats/GroupChatsTab.tsx

```diff
@@ -1,35 +1,43 @@
 import { Box } from "@material-ui/core";
 import { makeStyles } from "@material-ui/core/styles";
-import React, { useEffect } from "react";
-import { fetchGroupChats } from ".";
+import React, { useState } from "react";
+import { useQuery } from "react-query";
 import Alert from "../../../components/Alert";
 import CircularProgress from "../../../components/CircularProgress";
-import { useAppDispatch, useTypedSelector } from "../../../store";
+import { Error as GrpcError } from "grpc-web";
+import { GroupChat } from "../../../pb/conversations_pb";
 import CreateGroupChat from "./CreateGroupChat";
 import GroupChatList from "./GroupChatList";
 import GroupChatView from "./GroupChatView";
+import { service } from "../../../service";
 
 const useStyles = makeStyles({ root: {} });
 
 export default function GroupChatsTab() {
-  const dispatch = useAppDispatch();
-  const state = useTypedSelector((state) => state.groupChats);
+  const classes = useStyles();
 
-  useEffect(() => {
-    dispatch(fetchGroupChats());
-  }, [dispatch]);
+  ///TODO: add url chat opening
+
+  const [groupChat, setGroupChat] = useState<GroupChat.AsObject | null>(null);
+
+  /// TODO: Pagination
+  const { data: groupChats, isLoading, error } = useQuery<
+    GroupChat.AsObject[],
+    GrpcError
+  >(["groupChats"], () => service.conversations.listGroupChats());
+
+  const closeGroupChat = () => setGroupChat(null);
 
-  const classes = useStyles();
   return (
     <Box className={classes.root}>
-      {state.error && <Alert severity={"error"}>{state.error}</Alert>}
-      {state.loading ? (
+      {error && <Alert severity={"error"}>{error.message}</Alert>}
+      {isLoading ? (
         <CircularProgress />
-      ) : state.groupChatView.groupChat ? (
-        <GroupChatView />
+      ) : groupChat ? (
+        <GroupChatView groupChat={groupChat} closeGroupChat={closeGroupChat} />
       ) : (
         <>
-          <GroupChatList />
+          <GroupChatList groupChats={groupChats!} setGroupChat={setGroupChat} />
```

Comment:
I assume the API returns an empty list if no group chats exist yet for the current user?

Reply:
Yes, does that mean ! will cause problems, or is it fine?

Reply:
It might be problematic if there's a data loading error since we'll be trying to map over `undefined` in `GroupChatList`. Probably worth marking the `groupChats` prop as optional and deal with it in `GroupChatList`?